src/throttler.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { ThrottlerStorage } from './throttler-storage.interface';
+
+@Injectable()
+export class ThrottlerStorageService implements ThrottlerStorage {
+  storage: Record<string, number[]> = {};
+
+  async getRecord(key: string): Promise<number[]> {
+    return this.storage[key] || [];
+  }
+
+  async addRecord(key: string, ttl: number): Promise<void> {
+    const ttlMilliseconds = ttl * 1000;
+    if (!this.storage[key]) {
+      this.storage[key] = [];
+    }
+
+    this.storage[key].push(Date.now() + ttlMilliseconds);
+
+    const timeoutId = setTimeout(() => {
+      this.storage[key].shift();
+      clearTimeout(timeoutId);
+    }, ttlMilliseconds);
```

Comment:
Should we store these timeout handlers and close them immediately on the `onApplicationShutdown` hook?

Reply:
Sounds like it would be a good idea. Got any examples I could look at?

Reply:
Would it just be as simple as something like this:
```ts
async addRecord(key: string, ttl: number): Promise<void> {
    const ttlMilliseconds = ttl * 1000;
    if (!this.storage[key]) {
      this.storage[key] = [];
    }

    this.storage[key].push(Date.now() + ttlMilliseconds);

    const timeoutId = setTimeout(() => {
      this.storage[key].shift();
      clearTimeout(timeoutId);
    }, ttlMilliseconds);
    this.timeoutIds.push(timeoutId);
  }

  onApplicationShutdown() {
    this.timeoutIds.forEach(clearTimeout);
  }
```